instrumentation/opentelemetry/init.go

```diff
@@ -59,37 +77,80 @@ func Init(cfg *config.AgentConfig) func() {
 		log.Fatal(err)
 	}
 
+	defaultBatcher := sdktrace.NewBatchSpanProcessor(zipkinExporter)
+
 	resources, err := resource.New(
 		context.Background(),
-		resource.WithAttributes(makeResources(cfg)...),
+		resource.WithAttributes(createResources(cfg.GetServiceName().GetValue(), cfg.ResourceAttributes)...),
 	)
 	if err != nil {
 		log.Fatal(err)
 	}
-
+	defaultSampler := sdktrace.AlwaysSample()
 	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
-		sdktrace.WithBatcher(zipkinBatchExporter, sdktrace.WithBatchTimeout(batchTimeout)),
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: defaultSampler}),
+		sdktrace.WithSpanProcessor(defaultBatcher),
 		sdktrace.WithResource(resources),
 	)
 	otel.SetTracerProvider(tp)
 
 	otel.SetTextMapPropagator(makePropagator(cfg.PropagationFormats))
 
+	traceProviders = make(map[string]*sdktrace.TracerProvider)
+	batcher = defaultBatcher
+	sampler = defaultSampler
+	initialized = true
 	return func() {
+		mu.Lock()
+		defer mu.Unlock()
+		batcher.Shutdown(context.Background())
+		for _, tracerProvider := range traceProviders {
+			tracerProvider.Shutdown(context.Background())
```

Comment:
I think you can also delete the traceprovider from the map.

Reply:
Done.